numba/core/overload_glue.py

```diff
@@ -183,16 +199,36 @@ def __call__(self, func, typing_key=None):
 del _Gluer
 
 
+_no_defer = set()
+
+
 def glue_typing(concrete_function, typing_key=None):
     """This is a decorator for wrapping the typing part for a concrete function
     'concrete_function', it's a text-only replacement for '@infer_global'"""
     return _overload_glue(concrete_function,
                           typing_key=typing_key).wrap_typing()
 
 
+_deferred_lowering = defaultdict(list)
+
+
```

Comment:
Am wondering if having these in globals is going to lead to race conditions later?